File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/DecksPresenter.java
package com.dbottillo.mtgsearchfree.presenter;

import android.net.Uri;

import com.dbottillo.mtgsearchfree.model.Deck;
import com.dbottillo.mtgsearchfree.model.MTGCard;
import com.dbottillo.mtgsearchfree.view.DecksView;

import java.util.List;

public interface DecksPresenter {

    void init(DecksView view);

    void loadDecks();

    void loadDeck(Deck deck);

    void addDeck(String name);

    void deleteDeck(Deck deck);

    void editDeck(Deck deck, String name);

    void addCardToDeck(Deck deck, MTGCard card, int quantity);

    void addCardToDeck(String name, MTGCard card, int quantity);

    void removeCardFromDeck(Deck deck, MTGCard card);

    void removeAllCardFromDeck(Deck deck, MTGCard card);

    void importDeck(Uri uri);

    void exportDeck(Deck deck, List<MTGCard> cards);
}

File: MTGSearch/src/main/java/com/dbottillo/mtgsearchfree/presenter/DecksPresenter.kt
package com.dbottillo.mtgsearchfree.presenter

import android.net.Uri

import com.dbottillo.mtgsearchfree.model.CardsCollection
import com.dbottillo.mtgsearchfree.model.Deck
import com.dbottillo.mtgsearchfree.model.MTGCard
import com.dbottillo.mtgsearchfree.view.DecksView

interface DecksPresenter {

    fun init(view: DecksView)

    fun loadDecks()

    fun loadDeck(deck: Deck)

    fun addDeck(name: String)

    fun deleteDeck(deck: Deck)

    fun editDeck(deck: Deck, name: String)

    fun addCardToDeck(deck: Deck, card: MTGCard, quantity: Int)

    fun addCardToDeck(name: String, card: MTGCard, quantity: Int)

    fun removeCardFromDeck(deck: Deck, card: MTGCard)

    fun removeAllCardFromDeck(deck: Deck, card: MTGCard)

    fun moveCardFromSideBoard(deck: Deck, card: MTGCard, quantity: Int)

    fun moveCardToSideBoard(deck: Deck, card: MTGCard, quantity: Int)

    fun importDeck(uri: Uri)

    fun exportDeck(deck: Deck, cards: CardsCollection)
}